gradle-consistent-versions-idea-plugin/src/test/java/com/palantir/gradle/versions/intellij/RepositoryExplorerTests.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+public class RepositoryExplorerTests {
+
+    @Test
+    void test_gets_release_from_metadata() {
+        RepositoryExplorer repositoryExplorer = new RepositoryExplorer();
+
+        Versioning versioning = ImmutableVersioning.builder()
```

Comment:
Do not use `Immutable*` types outside the immutables classes. Make builds for them that returns the no `Immutable*` type (ie `static Versioning builder()`)

Reply:
So I've set this up to work like in GCV where we return a builder, is this the correct thing todo? 

```java
    static Builder builder() {
        return new Builder();
    }

    class Builder extends ImmutableVersioning.Builder {}
```